src/fleetcommanderclient/configadapters/networkmanager.py

```diff
@@ -94,14 +84,32 @@ class NetworkManagerConfigAdapter(BaseConfigAdapter):
     def bootstrap(self, uid):
         self.nmhelper = NetworkManagerDbusHelper()
 
+    def add_connection_metadata (self, serialized_data, uname, conn_uuid):
+        sc = NM.SimpleConnection.new_from_dbus(GLib.Variant.parse(None, serialized_data, None, None))
+        setu = sc.get_setting(NM.SettingUser)
+        if not setu:
+            sc.add_setting(NM.SettingUser())
+            setu = sc.get_setting(NM.SettingUser)
+
+        setc = sc.get_setting(NM.SettingConnection)
+
+        setu.set_data('org.fleet-commander.connection', 'true')
+        setu.set_data('org.fleet-commander.connection.uuid', conn_uuid)
+        setc.uuid = uuid.uuid5(uuid.UUID(conn_uuid), uname)
+        setc.add_permission ("user", uname, None)
+
+        return (sc.to_dbus(NM.ConnectionSerializationFlags.NO_SECRETS), setc.uuid)
+
     def update(self, uid, data):
+        uname = self.nmhelper.get_user_name(uid)
         for connection in data:
-            uuid = connection['uuid']
-            connection_data = self.nmhelper.variant_parse(connection['data'])
```

Comment:
Seen you have removed self.variant_parse calls.

I had to add them last time because I was not able to send/read the data through dbus without parsing the variant data. If this worked to you that's good news :+1: